src/java/org/apache/cassandra/transport/PreV5Handlers.java

```diff
@@ -123,48 +129,113 @@ private void releaseItem(Flusher.FlushItem<Message.Response> item)
          * purpose since it does not have the message envelope associated with the exception.
          * <p>
          * Note: this method should execute on the netty event loop.
+         * 
+         * @return the type of {@link Overload} triggered
          */
-        private boolean shouldHandleRequest(ChannelHandlerContext ctx, Message.Request request)
+        private Overload checkLimits(ChannelHandlerContext ctx, Message.Request request)
```

Comment:
Consider mentioning in javadoc comment here why some of the paths set back pressure and others don't. It's not clear at face value w/out tracing the usage of the return why some conditional branches set it and others don't, which is liable to cause issues in the future if someone comes in here to modify it and there's that subtlety.

Reply:
Yeah, it's purpose was basically to communicate to the dispatcher whether or not we need to issue a client warning without changing the existing plumbing for enabling their collection. Agreed that a comment would clarify this though.